dbms/src/Common/SensitiveDataMasker.h

```diff
@@ -0,0 +1,173 @@
+
+
+#ifndef NDEBUG
+#    include <iostream>
+#endif
+
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+#include <re2/re2.h>
+#include <re2/stringpiece.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <Common/Exception.h>
+#include <Common/StringUtils/StringUtils.h>
+#include <common/logger_useful.h>
+
+
+//* TODO: option with hyperscan? https://software.intel.com/en-us/articles/why-and-how-to-replace-pcre-with-hyperscan
+// re2::set should also work quite fast, but it doesn't return the match position, only which regexp was matched
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPILE_REGEXP;
+    extern const int NO_ELEMENTS_IN_CONFIG;
+    extern const int INVALID_CONFIG_PARAMETER;
+
+}
+
+
+class SensitiveDataMasker
+{
+private:
+    class MaskingRule
+    {
+    private:
+        const std::string name;
+        const std::string replacement_string;
+        const std::string regexp_string;
+
+        const RE2 regexp;
+        const re2::StringPiece replacement;
+
+        std::atomic<std::uint64_t> matches_count = 0;
```

Comment:
BTW, we can make it "mutable" and make "apply" method const. Because it's a good practice to have thread-safe method const.